File: library/src/main/java/com/getstream/sdk/chat/navigation/destinations/WebLinkDestination.java
package com.getstream.sdk.chat.navigation.destinations;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;

import com.getstream.sdk.chat.Chat;

public class WebLinkDestination extends ChatDestination {

    public final String url;

    public WebLinkDestination(String url, Context context) {
        super(context);
        String signedUrl = Chat.getInstance().urlSigner().signFileUrl(url);
        this.url = signedUrl == null ? "" : signedUrl;
    }

    @Override
    public void navigate() {
        Intent browserIntent = new Intent(
                Intent.ACTION_VIEW,
                Uri.parse(url)
        );
        start(browserIntent);
    }
}

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/navigation/destinations/WebLinkDestination.kt
package com.getstream.sdk.chat.navigation.destinations

import android.content.Context
import android.content.Intent
import android.net.Uri
import com.getstream.sdk.chat.UrlSigner
import io.getstream.chat.android.core.internal.InternalStreamChatApi

@InternalStreamChatApi
public class WebLinkDestination(url: String, context: Context, urlSigner: UrlSigner = UrlSigner.DefaultUrlSigner()) :
    ChatDestination(context) {

    private val url: String = urlSigner.signFileUrl(url) ?: ""

    override fun navigate() {
        val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
        start(browserIntent)
    }
}
